### ISSUE TITLE
mathgl[qt5]:x64-windows build failure

### ISSUE THREAD

**Issue by spindensity** [microsoft/vcpkg#8360]
**Host Environment**
- OS: Windows 10 1903
- MSVC 2019 16.3.0

**To Reproduce**
```
vcpkg install mathgl[qt5]:x64-windows
```

**Failure logs** 
[config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/3658271/config-x64-windows-out.log)
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/3658272/install-x64-windows-dbg-out.log)

**Comment by Neumann-A** [comment]
Error:
```
mgl-staticd.lib(parser.cpp.obj) : error LNK2005: mgl_ask_func already defined in mglview.cpp.obj
mgl-staticd.lib(parser.cpp.obj) : error LNK2005: mgl_progress_func already defined in mglview.cpp.obj
mgl-staticd.lib(canvas.cpp.obj) : error LNK2005: "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > mglGlobalMess" (?mglGlobalMess@@3V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@A) already defined in mglview.cpp.obj
utils\mglview.exe : fatal error LNK1169: one or more multiply defined symbols found
```

remove part of the patch vcpkg adds in fix_cmakelists_and_cpp.patch and retry: 
```
diff --git a/utils/mglview.cpp b/utils/mglview.cpp
index d360df3..15eeb3e 100644
--- a/utils/mglview.cpp
+++ b/utils/mglview.cpp
@@ -29,6 +29,12 @@
 	#include "mgl2/qt.h"
 #endif
 //-----------------------------------------------------------------------------
+#if WIN32
+	MGL_EXPORT void (*mgl_ask_func)(const wchar_t *, wchar_t *)=0;
+	MGL_EXPORT void (*mgl_progress_func)(int value, int maximal, HMGL)=0;
+	MGL_EXPORT std::string mglGlobalMess;
+#endif
+//-----------------------------------------------------------------------------
 std::wstring str, opt;
 std::vector<std::string> anim;
 mglParse p(true);
```

**Comment by JackBoosY** [comment]
After searching for `mgl_abs`, I found that this function was removed in a certain version, and we should report this to the official.
See [history](https://fossies.org/diffs/mathgl/2.4.2.1_vs_2.4.3/include/mgl2/define.h-diff.html).

**Comment by Neumann-A** [comment]
@JackBoosY wrong issue: you meant to reply to #8357

**Comment by JackBoosY** [comment]
OMG... reply in wrong issue.

**Comment by JackBoosY** [comment]
Hi @spindensity, thanks for reporting this issue.
I've fixed this issue in #8369, please update vcpkg and rebuild mathgl.

Thanks.
